### DISCUSSION TITLE
Maybe the Pagination  component is needed

### DISCUSSION THREAD

**Discussion by tiyee** [segmentio/evergreen#1259] [Ideas] [answered]
as the title said ~

**Comment by mshwery** [comment]
Hey @tiyee ! Can you elaborate on what your use case is? What type of pagination? How would you envision consuming it?

We've definitely thought about whether this belongs in Evergreen, or not. Interested in hearing your thoughts!

**Comment by tiyee** [comment]
> Hey @tiyee ! Can you elaborate on what your use case is? What type of pagination? How would you envision consuming it?
> 
> We've definitely thought about whether this belongs in Evergreen, or not. Interested in hearing your thoughts!

it is a standard component of data-view page.for example, in your blog article list page,it can not show all the articles in one page,so you need pagination.

How to design ? maybe could learn from others' practices ,like [ant.design/components/pagination/](https://ant.design/components/pagination/)

**Comment by mshwery** [comment]
Makes sense.

I think this type of pagination fits within the theme of Evegreen, it's basically one step higher than atomic components – since it's a composition of several components but isn't that large or opinionated.

If you have ideas, would love to see them in a PR.

**Comment by mshwery** [comment, marked as answer]
There is a https://evergreen.segment.com/components/pagination now!

**Reply by tiyee** [reply]
well done! thanks!

**Reply by MMMeiyappan** [reply]
hey yes, it's now available, but it still has a lot of bare-bones code in it,
and it doesn't work the way it should.
when we click on one of the page numbers, it goes to that but the page numbers around it don't show up.

**Reply by brandongregoryscott** [reply]
@MMMeiyappan Please open up an issue with a code sample/sandbox of how you're using the component to further debug, thanks!
